libpod/container_internal_linux.go

```diff
@@ -1098,6 +1100,160 @@ func (c *Container) addNamespaceContainer(g *generate.Generator, ns LinuxNS, ctr
 	return nil
 }
 
+func (c *Container) addCheckpointImageMetadata(importBuilder *buildah.Builder) error {
+	// Get information about host environment
+	hostInfo, err := c.Runtime().hostInfo()
+	if err != nil {
+		return fmt.Errorf("getting host info: %v", err)
+	}
+
+	criuVersion, err := criu.GetCriuVestion()
+	if err != nil {
+		return fmt.Errorf("getting criu version: %v", err)
+	}
+
+	// Add image annotations with information about the container and the host.
+	// This information is useful to check compatibility before restoring the checkpoint
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationName,
+		c.Config().Name,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationRawImageName,
+		c.Config().RawImageName,
+	)
+
+	rootfsImageID, rootfsImageName := c.Image()
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationRootfsImageID,
+		rootfsImageID,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationRootfsImageName,
+		rootfsImageName,
+	)
+
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationPodmanVersion,
+		version.Version.String(),
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationCriuVersion,
+		strconv.Itoa(criuVersion),
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationRuntimeName,
+		hostInfo.OCIRuntime.Name,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationRuntimeVersion,
+		hostInfo.OCIRuntime.Version,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationConmonVersion,
+		hostInfo.Conmon.Version,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationHostArch,
+		hostInfo.Arch,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationHostKernel,
+		hostInfo.Kernel,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationCgroupVersion,
+		hostInfo.CgroupsVersion,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationDistributionVersion,
+		hostInfo.Distribution.Version,
+	)
+	importBuilder.SetAnnotation(
+		define.CheckpointAnnotationDistributionName,
+		hostInfo.Distribution.Distribution,
+	)
+
+	return nil
+}
+
+func (c *Container) validateCheckpointImageName(options *ContainerCheckpointOptions) error {
```

Comment:
```suggestion
func (c *Container) resolveCheckpointImageName(options *ContainerCheckpointOptions) error {
```
Nit: no real validation is happening